sprox/sa/support.py

```diff
@@ -32,5 +32,7 @@ def resolve_entity(entity):
         entity = entity()
     if _class_resolver is not None and isinstance(entity, _class_resolver):
         entity = entity()
+    elif hasattr(entity, '__self__'):
+        entity = entity()
```

Comment:
Sounds reasonable, but probably extending the original `if inspect.isfunction` to `if inspect.isfunction or inspect.ismethod` would probably achieve the same in a way that feels more expressive about what was the intent.

Reply:
That works. Made the change and tested in context where issue was observed. Thanks.